sdk/python/feast/data_source.py

```diff
@@ -875,3 +909,223 @@ def to_proto(self) -> DataSourceProto:
         data_source_proto.date_partition_column = self.date_partition_column
 
         return data_source_proto
+
+
+class RedshiftOptions:
+    """
+    DataSource Redshift options used to source features from Redshift query
+    """
+
+    def __init__(self, table: Optional[str], query: Optional[str]):
+        self._table = table
+        self._query = query
+
+    @property
+    def query(self):
```

Comment:
Do we need a getter/setter if we aren't adding custom logic?

Reply:
Yeah... We use the exact same approach for most other fields in DataSource/FileSource/BigQuerySource though. I think it would be good to clean this up separately. wdyt?

Reply:
sgtm